src/v/config/property.h

```diff
@@ -123,10 +131,180 @@ class property : public base_property {
     };
 };
 
+template<typename T>
+requires((std::is_integral_v<T>)&&(
+  !std::is_same_v<T, const bool>)) std::string_view property_type_name(T*) {
+    return "integer";
+}
```

Comment:
Thanks for the pointers here, it was my first time with concepts.  I've revised this to a more concise version with inline `if` blocks (apart from property_units_name, which was triggering a segfault in clang, eek)

The static_assert was a bit troublesome, because it generates just a static string error, with no info about which type's instantiation of the template caused the issue.  Instead, I return `""` on the unhandled path, and use a wrapper function with a `requires` that this value isn't returned -- that way if an unhandled property type is used, the compiler error gives an error that points to which type is the problem.